src/main/scala/is/hail/expr/AST.scala

```diff
@@ -236,7 +236,7 @@ sealed abstract class AST(pos: Position, subexprs: Array[AST] = Array.empty) {
   def runAggregator(ec: EvalContext): CPS[Any] = {
     val typedNames = ec.st.toSeq
       .sortBy { case (_, (i, _)) => i }
-      .map { case (name, (_, typ)) => (name, typ) }
+      .map { case (name, (_, typ)) => (name, typ.scalaClassTag.asInstanceOf[ClassTag[AnyRef]]) }
```

Comment:
Scala has `forSome` for existential types, so you don't need to cook up your own `Existential.

 > in Scala, the existential is bound at the edges of the type expression

Can we double-check this?

```
scala> val b: mutable.Map[String, Array[T] forSome { type T }] = mutable.Map("foo" -> Array(1, 2, 3))
b: scala.collection.mutable.Map[String,Array[_]] = Map(foo -> Array(1, 2, 3))
```

Note the printed type, vs:

```
scala> val c: mutable.Map[String, Array[T]] forSome { type T } = mutable.Map("foo" -> Array(1, 2, 3))
c: scala.collection.mutable.Map[String,Array[T]] forSome { type T } = Map(foo -> Array(1, 2, 3))
```

And they behave as expected:

```
scala> b += "bar" -> Array("baz")
res2: b.type = Map(foo -> Array(1, 2, 3), bar -> Array(baz))

scala> c += "bar" -> Array("baz")
<console>:14: error: type mismatch;
 found   : String("baz")
 required: T
       c += "bar" -> Array("baz")
```

Also:

```
scala> val d: mutable.Map[String, Array[_]] = mutable.Map("foo" -> Array(1, 2, 3))
d: scala.collection.mutable.Map[String,Array[_]] = Map(foo -> Array(1, 2, 3))

scala> d += ("bar" -> Array("baz"))
res5: d.type = Map(foo -> Array(1, 2, 3), bar -> Array(baz))
```